great_expectations/render/renderer/new_site_builder.py

```diff
@@ -0,0 +1,253 @@
+
+            if self.run_id_filter:
+                if not self._resource_key_passes_run_id_filter(resource_key):
+                    continue
+
+            resource = self.source_store.get(resource_key)
+
+            # TODO : This will need to change slightly when renderer and view classes are configurable.
+            # TODO : Typing resources is SUPER important for usability now that we're slapping configurable renders together with arbitrary stores.
+            rendered_content = self.renderer_class.render(resource)
+            viewable_content = self.view_class.render(rendered_content)
+
+            self.target_store.set(
+                SiteSectionIdentifier(
+                    site_section_name=self.name,
+                    resource_identifier=resource_key,
+                ),
+                viewable_content
+            )
+
+            #Where do index_link_dicts live?
+            # In the HtmlSiteStore!
+    
+    def _resource_key_passes_run_id_filter(self, resource_key):
+        if type(resource_key) == ValidationResultIdentifier:
+            run_id = resource_key.run_id
+        else:
+            raise TypeError("run_id_filter filtering is only implemented for ValidationResultResources.")
+
+        if self.run_id_filter.get("eq"):
+            return self.run_id_filter.get("eq") == run_id
+
+        elif self.run_id_filter.get("ne"):
+            return self.run_id_filter.get("ne") != run_id
+
+
+class DefaultSiteIndexBuilder(object):
+
+    def __init__(self,
+        name,
+        data_context,
+        target_store_name,
+        renderer=None,
+        view=None,
+    ):
+        # NOTE: This method is almost idenitcal to DefaultSiteSectionBuilder
+        self.name = name
+
+        self.target_store = data_context.stores[target_store_name]
+
+        # TODO : Push conventions for configurability down to renderers and views.
+        # Until then, they won't be configurable, and defaults will be hard.
+        if renderer == None:
+            renderer = {
+                "module_name": "great_expectations.render.renderer",
+                "class_name": "SiteIndexPageRenderer",
+            }
+        renderer_module = importlib.import_module(renderer.pop("module_name"))
+        self.renderer_class = getattr(renderer_module, renderer.pop("class_name"))
+
+
+        if view == None:
+            view = {
+                "module_name" : "great_expectations.render.view",
+                "class_name" : "DefaultJinjaIndexPageView",
+            }
+        view_module = importlib.import_module(view.pop("module_name"))
+        self.view_class = getattr(view_module, view.pop("class_name"))
+
+
+    def build(self):
+        # Loop over sections in the HtmlStore
+        logger.debug("DefaultSiteIndexBuilder.build")
+
+        resource_keys = self.target_store.list_keys()
+
+        # FIXME : There were no tests to verify that content is created or correct,
```

Comment:
@roblim These five lines are the place where I need some help. To activate these, we'll need to refactor `SiteIndexPageRenderer`